mod/admin.php

```diff
@@ -290,7 +290,7 @@ function admin_page_federation(&$a) {
 		// what versions for that platform do we know at all?
 		// again only the active nodes
 		$v = qu('SELECT COUNT(*) AS `total`, `version` FROM `gserver`
-				WHERE `last_contact` > `last_failure` AND `platform` LIKE "%s"  AND `version` != ""
+				WHERE `last_contact` > `last_failure` AND `platform` LIKE "%s"  AND `version` RLIKE "^[[:digit:]]+[-.][[:digit:]]+"
```

Comment:
I think cutting does not change anything. There should be a filter that filter and count version numbers. The statistics are wrong anyway, because empty strings are always excluded. I think I withdraw my pull request and work on a more general solution, i.e. counting empty version numbers as "unknown", excluding error messages and find some infos how version numbers looks like and fixing the total node counters too.